sdk/src/main/scala/com/horizen/websocket/client/MainchainNodeChannelImpl.scala

```diff
@@ -27,6 +27,15 @@ case class SendCertificateRequestPayload(scid: String,
                                          vFieldElementCertificateField: Seq[String], // Seq of compressed FE hex strings
                                          vBitVectorCertificateField: Seq[String] // Seq of compressed bitvectors hex strings
                                         ) extends RequestPayload
+case class SendCertificateAutoFeeRequestPayload(scid: String,
```

Comment:
Do we need 2  different classes: `SendCertificateRequestPayload` and `SendCertificateAutoFeeRequestPayload`?
I would expect, if in `SendCertificateRequestPayload` we change `fee` type from `String` to `Option[String]` it will work for us in any case. So, in case there is auto fee -> `fee == None` -> field will not be serialized to JSON.

Reply:
Mainchain after receiving send_certificate websocket command process arguments sequently. In case we don't send fee, we also shouldn't pass vFieldElementCertificateField and  vBitVectorCertificateField, because the information in these fields will be processed as fee by Mainchain.

Reply:
as agreed, some changes on MC side need to be done. Then we will update SDK code.